### PR TITLE
Guide on AutoDl

### PR DESCRIPTION
**Pull request by SurajBaloni**
Guide for AutoDL API in arcgis.learn

### CODE REVIEW

**Comment by mohi9282** [comment]
@SurajBaloni - 2.0.1 release is coming up fast. If you would like this sample/guide to go in the release, please add the milestone to this PR and have it reviewed, and approved by 18th May.

File: guide/14-deep-learning/how_AutoDL_works.ipynb [hunk 1]
```diff
@@ -0,0 +1,566 @@
+{
```
Comment (KarthikDutt) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p>Reword:</p><p>This guide explains the steps for training and evaluating multiple network performance supported by <code>arcgis.learn</code> API. </p><p>to </p><p>This guide explains the steps for training and evaluation of multiple network architectures supported by <code>arcgis.learn</code> API.</p><p> </p><p>comma is needed after <code>more specifically for object detetcion task only</code></p><p> </p><p>comma is needed after 'train a deep learning network using the <code>arcgis.learn</code> API'</p><p> </p><p>remove 'and' from <code>AutoDL class automatically trains all the supported network with the given data and</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (KarthikDutt) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p>I think it is better to say 'Prepare data for' instead of 'Prepare databunch for'</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (KarthikDutt) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p><code>The total time limit in hours for AutoDL to train the networks.</code> to <code>The total time limit in hours for AutoDL to train and evaluate the networks.</code>   </p><p> </p><p>Consider renaming  'perform' to 'Advanced' to be similar to AutoML. </p><p> </p><p><code>To be used when the user wants to tune hyper parameters of two best performing models from basic mode.</code> can be changed to <code>To be used when the user additionally wants to tune hyper parameters of two best performing models from basic mode.</code>  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (KarthikDutt) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p><code>however if the remaining time left to process the network is less than than the expected time</code> to </p><p><code>however if the remaining time left to process the network is less than the expected time</code> </p><p>Can you clarify what expected time means? </p><p> </p><p>We can reword this statement <code>In the second half it checks for two best performing networks.</code></p><p> </p><p><code>than than the expected time</code> . Maybe we can reword   <code>expected </code> time</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (KarthikDutt) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p>We can reword <code>When you initialize the AutoDL class it calculates</code> to <code>When AutoDL class is initialized, it calculates</code>  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (KarthikDutt) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p><code>This method will return a dataframe</code> to </p><p><code>This method will return a evaluation report as a dataframe that will include several fields including model's accuracy </code>  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Comment (KarthikDutt) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p>We can change this heading to <code>How Imagery Model can be used to improve the results from AutoDL</code> or <code>Improve results from AutoDL using ImageryModel</code>  or something similar</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

Comment (KarthikDutt) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p><code>this class supports method that</code> to <code>this class supports methods that</code> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (KarthikDutt) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p>Here, we can add a summary to briefly describe what we did in this notebook. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='57'/>

**Comment by priyankatuteja** [comment]
@BP-Ent could you help in reviewing it

File: guide/14-deep-learning/how_AutoDL_works.ipynb [hunk 2]
```diff
@@ -0,0 +1,580 @@
...
```
Comment (BP-Ent) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p>The output of this function will be a list of pixel classification models supported by the <code>AutoDL</code> class.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (BP-Ent) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p>The output of this function will be a list of object detection models supported by the <code>AutoDL</code> class.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

Comment (BP-Ent) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p>The <code>fit</code> method will be used to train all of the selected networks automatically within the provided time limit.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (BP-Ent) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p>This method will return an evaluation report as a dataframe that will include several fields, including the model's accuracy with train/validation loss, dice(for pixel classification), the learning rate used to train the model, train time, and backbone.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Comment (BP-Ent) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p>This method will display the results for the best performing model.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Comment (BP-Ent) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p>Once the best performing network is identified, it can be further fine tuned using the <code>ImageryModel</code> class. This class supports methods that can be used to load, fine-tune, and save the model for further use.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (BP-Ent) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <ul><li>The <code>load</code>&nbsp;method is used to load a saved model from the disk using the&nbsp;<code>AutoDL</code>&nbsp;class. It accepts the following parameters:</li><li class="ql-indent-1">path: Path to the ESRI Model Definition (.emd) file</li><li class="ql-indent-1">data: Returned data object from <code>prepare_data</code> function</li></ul><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (BP-Ent) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <h3>Learning rate and loading the model</h3><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='48'/>

Comment (BP-Ent) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p>The <code>lr_find</code> method runs the Learning Rate Finder, which helps in choosing the optimum learning rate for training the model.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='49'/>

Comment (BP-Ent) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p>This method saves the model weights and creates an Esri Model Definition and a Deep Learning Package zip for deployment to Image Server or ArcGIS Pro.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='55'/>

Comment (BP-Ent) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p>This guide has explained how the <code>AutoDL</code> class can be used to automate multiple deep learning models supported by the<code>arcgis.learn</code> API. For every step in the workflow, we defined a function and discussed its usage. This guide can be a starting point for developers to train and evaluate multiple <code>arcgis.learn</code> supported models' performances.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='58'/>

Comment (BP-Ent) [inline guide/14-deep-learning/how_AutoDL_works.ipynb:1]: <p>For more information about the API, refer to the <a href="https://developers.arcgis.com/python/api-reference/arcgis.learn.toc.html#autodl" target="_blank">API reference</a>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='59'/>

**Review by BP-Ent — DISMISSED** [review]
Suggested changes made on reviewnb

**Review by priyankatuteja — APPROVED** [review]
